java/arcs/core/analysis/ExpressionDependencyAnalyzer.kt

```diff
@@ -26,25 +26,27 @@ private typealias Scope = DependencyNode.AssociationNode
  * target [Expression].
  */
 class ExpressionDependencyAnalyzer : Expression.Visitor<DependencyNode, Scope> {
```

Comment:
Shouldn't the visitor return a pair of `DependencyNode`, where one would correspond to data dependence nodes and the other would correspond to influence nodes?

I think we would need to return both data dependence and the influence nodes for the following case?
```
from f in foo
let x = f.a + f.b
from g in bar
let y = g.x
select new Blah {res: y + x}
```

Reply:
Can you explain how this example would need influence tracking? It does not have any `where` expressions. In any case, I see what you mean, and have an answer: Instead of having a pair of `DependencyNode`s, I use the `BufferedScope` data class (this is an answer to https://github.com/PolymerLabs/arcs/pull/6547/files#r523270530).